pyqtgraph/opengl/items/GLTextItem.py

```diff
@@ -0,0 +1,107 @@
+from OpenGL.GL import *
+import numpy as np
+from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+
+class GLTextItem(GLGraphicsItem):
+    """Draws text in 3D."""
+
+    def __init__(self, **kwds):
+        """All keyword arguments are passed to setData()"""
+        GLGraphicsItem.__init__(self)
+        glopts = kwds.pop('glOptions', 'additive')
+        self.setGLOptions(glopts)
+
+        self.pos = np.array([0.0, 0.0, 0.0])
+        self.color = QtCore.Qt.white
+        self.text = ''
+        self.font = QtGui.QFont('Helvetica', 16)
+
+        self.setData(**kwds)
+    
+    def setData(self, **kwds):
+        """
+        Update the data displayed by this item. All arguments are optional;
+        for example it is allowed to update text while leaving colors unchanged, etc.
+
+        ====================  ==================================================
+        **Arguments:**
+        ------------------------------------------------------------------------
+        pos                   (3,) array of floats specifying text location.
+        color                 QColor or array of ints [R,G,B] or [R,G,B,A]. (Default: Qt.white)
+        text                  String to display.
+        font                  QFont (Default: QFont('Helvetica', 16))
+        ====================  ==================================================
+        """
+        args = ['pos', 'color', 'text', 'font']
+        for k in kwds.keys():
+            if k not in args:
+                raise Exception('Invalid keyword argument: %s (allowed arguments are %s)' % (k, str(args)))
+        for arg in args:
+            if arg in kwds:
+                value = kwds[arg]
+                if arg == 'pos':
+                    if isinstance(value, np.ndarray):
+                        if value.shape != (3,):
+                            raise Exception('"pos.shape" must be (3,).')
+                    elif isinstance(value, (tuple, list)):
+                        if len(value) != 3:
+                            raise Exception('"len(pos)" must be 3.')
+                elif arg == 'color':
```

Comment:
I think you can get rid of all your logic here, and just say:

```python
from pyqtgraph import functions as fn
...


value = fn.mkColor(value)
```

We have a lot of logic here to support a _wide_ variety of inputs, and we raise the appropriate exceptions; end result should be what you want.

Reply:
I fixed in [8946076](https://github.com/pyqtgraph/pyqtgraph/pull/1776/commits/8946076752840caf734c783e1d05c330702620c6).